### PR TITLE
add  DUP17-32 and SWAP17-32 instructions to expand local variable number

### PR DESCRIPTION
**Pull request by aion-camus**
relax the local variable limit, support up to 32 items in local stack.

### CODE REVIEW

**Comment by aion-camus** [comment]
New revision number is named AION_V1 to let kernel know the current version of fvm. It should be under discussion for release schedule. Local variable contract for test is uploaded.

File: fastvm/libevmjit/Compiler.cpp [hunk 1]
```diff
@@ -566,13 +566,19 @@ void Compiler::compileBasicBlock(BasicBlock& _basicBlock, RuntimeManager& _runti
...
 			stack.dup(index);
 			break;
 		}
 
 		case Instruction::ANY_SWAP:
 		{
 			auto index = static_cast<size_t>(inst) - static_cast<size_t>(Instruction::SWAP1) + 1;
+			if ((m_rev < EVM_AION_V1) && (index > 16)) {
+				goto invalidInstruction;
+			}
```
Comment (yulongaion) [inline fastvm/libevmjit/Compiler.cpp]: I think this is problematic. `0x90` is `SWAP1` but interpreted as `DUP17` for old versions. Also, did you update `ANY_DUP` and `ANY_SWAP`?

Comment (yulongaion) [reply fastvm/libevmjit/Compiler.cpp]: I was wondering whether it would help if we use unallocated byte code rather than replacing old ones.

Comment (aion-camus) [reply fastvm/libevmjit/Compiler.cpp]: yeah, replacing original instructions needs more concern. That's a lesson from this issue. reassign  DUP17 = 0xb0, SWAP17 = 0xc0

**Comment by yulongaion** [comment]
LGTM

**Comment by aionick** [comment]
I'm good with this, though I would like to see the "proof" of the test contract you wrote (an actual JUnit test that compiles it and uses it).

**Comment by aion-camus** [comment]
It is tested under Rust Kernel. In Java kernel, I just use run_v1 in FastVM module to lauch the new fvm. Anyone who works on Java kernel interface could have a better solution.

**Comment by aionick** [comment]
Gotcha. I will be fixing up some old fvm tests and adding some new ones in the coming weeks for all of this avm integration, so I'll add some for this.

File: fastvm/include/evm.h [hunk 2]
```diff
@@ -449,6 +449,7 @@ enum evm_revision {
     EVM_BYZANTIUM = 4,
     EVM_AION = 5,
     EVM_CONSTANTINOPLE = 6,
+    EVM_AION_V1 = 7,
```
Comment (ywu123) [inline fastvm/include/evm.h]: I'd suggest we remove `EVM_CONSTANTINOPLE` becauses it's not implemented yet, and assign 6 to EVM_AION_V1.

**Pull request merged**